src/db_iter.h

```diff
@@ -96,17 +109,23 @@ class TitanDBIterator : public Iterator {
       std::unique_ptr<BlobFilePrefetcher> prefetcher;
       status_ = storage_->NewPrefetcher(index.file_number, &prefetcher);
       if (status_.IsCorruption()) {
```

Comment:
Seems vulnerable to me. Any other unexpected and unresolved corruption could cause an infinite loop. Should work as a temporary patch though.

Reply:
why infinite loop? Line 93 will break the loop in final

Reply:
No loop then. But we could do better than catching a generic error in future.